File: go/twoSumII.go
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted

package leetcode

func twoSumII(numbers []int, target int) []int {
	left := 0
	right := len(numbers) - 1

	for left < right {
		sum := numbers[left] + numbers[right]

		if sum < target {
			left++
			continue
		}

		if sum > target {
			right--
			continue
		}

		return []int{left + 1, right + 1}
	}

	return []int{}
}

File: python/src/twoSumII.py
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted

from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        while left < right:
            s = numbers[left] + numbers[right]
            if s < target:
                left += 1
                continue
            if s > target:
                right -= 1
                continue

            return [left + 1, right + 1]

        return []
